bentoml/_internal/frameworks/lightgbm.py

```diff
@@ -176,121 +166,96 @@ def save(
             params, lgb_train, num_boost_round=20, valid_sets=lgb_eval
         )
 
+        # `save` the booster to BentoML modelstore:
         tag = bentoml.lightgbm.save("my_lightgbm_model", gbm, booster_params=params)
-    """  # noqa
-    context: t.Dict[str, t.Any] = {
-        "framework_name": "lightgbm",
-        "pip_dependencies": [f"lightgbm=={get_pkg_version('lightgbm')}"],
-    }
+    """  # noqa: LN001
+
+    # Ensure that `model` is actually the Booster object, and not for example one of the scikit-learn wrapper objects.
+    if not isinstance(model, lgb.basic.Booster):
+        # Deal with a LightGBM issue (https://github.com/microsoft/LightGBM/issues/3014)
+        if not hasattr(model, "fitted_"):
+            model.fitted_ = True
+
+        try:
+            model = model.booster_
+        except AttributeError as e:
+            logger.error(
+                "Unable to obtain a lightgbm.basic.Booster from the specified model."
+            )
+            raise e
+
+    assert isinstance(
+        model, lgb.basic.Booster
+    ), "Unable to obtain a lightgbm.basic.Booster from the specified model."
```

Comment:
In the previous implementation of LightGBM support we accepted either one of these scikit-learn wrapper objects or the lightgbm Booster directly. In the former case, we'd serialize the object with `joblib.dump` and in the latter by directly calling `.save_model`. I'm now following the XGBoost implementation and am _only_ supporting directly saving the model with `model.save_model`. However, I still want to allow the user to provide an sklearn wrapper object, so this code simply extracts the Booster object from it.